redis/src/main/scala/com/avsystem/commons/redis/actor/RedisConnectionActor.scala

```diff
@@ -102,14 +105,23 @@ final class RedisConnectionActor(address: NodeAddress, config: ConnectionConfig)
     case Connect =>
       log.debug(s"Connecting to $address")
       doConnect()
-    case Connected(connection, remoteAddress, localAddress) =>
-      log.debug(s"Connected to Redis at $address")
-      //TODO: use dedicated retry strategy for initialization instead of reconnection strategy
-      new ConnectedTo(connection, localAddress, remoteAddress).initialize(config.reconnectionStrategy)
-      readInitSender.foreach(_ ! ReadAck)
-    case _: ConnectionFailed =>
-      log.error(s"Connection attempt to Redis at $address failed")
-      tryReconnect(retryStrategy, new ConnectionFailedException(address))
+    case Connected(connection, remoteAddress, localAddress, connectionId) =>
+      if (currentConnectionId.forall(_ != connectionId)) {
+        log.error(s"Received Connected for connection different than currently trying to establish")
+        connection ! CloseConnection(immediate = true)
+      } else {
+        //TODO: use dedicated retry strategy for initialization instead of reconnection strategy
```

Comment:
You miss `log.debug(s"Connected to Redis at $address")`, I guess it's an accident